apps/core/src/migration/util/convertMonitorToAppDefinition.ts

```diff
@@ -0,0 +1,144 @@
+import { DashboardDefinition } from '../../dashboards/entities/dashboard-definition.entity';
+import {
+  DashboardWidget,
+  DashboardWidgetType,
+} from '../../dashboards/entities/dashboard-widget.entity';
+import {
+  MonitorWidgetType,
+  MonitorMetric,
+  SiteWiseMonitorDashboardDefinition,
+} from './monitorDashboardDefinition';
+
+const defaultResolution = '1m';
+const defaultAggregationType = 'AVERAGE';
+
+/**
+ * Default Monitor size is 3x3 squares
+ * Default Application size is 84x52 pixels
+ */
+const pixelsPerWidthUnit = 28; // 84 / 3
+const pixelsPerHeightUnit = 17; // 52 / 3
+
+const convertType = (monitorChartType: string) => {
+  switch (monitorChartType) {
+    case MonitorWidgetType.LineChart:
+      return DashboardWidgetType.LineScatterChart;
+    default:
+      return DashboardWidgetType.LineScatterChart;
+  }
+};
+
+const convertHeight = (height: number) => {
+  return height * pixelsPerHeightUnit;
+};
+
+const convertWidth = (width: number) => {
+  return width * pixelsPerWidthUnit;
+};
+
+const convertX = (x: number) => {
+  return x * pixelsPerWidthUnit;
+};
+
+const convertY = (y: number) => {
+  return y * pixelsPerHeightUnit;
+};
+
+const convertResolution = (resolution?: string) => {
+  if (resolution) {
+    if (resolution === 'raw') {
+      return defaultResolution;
+    }
+    return resolution;
+  }
+  return defaultResolution;
+};
+
+const convertProperties = (monitorMetrics?: MonitorMetric[]) => {
+  const defaultProperties = {
+    symbol: {
+      style: 'filled-circle',
+    },
+    axis: {
+      yVisible: true,
+      xVisible: true,
+    },
+    line: {
+      connectionStyle: 'linear',
+      style: 'solid',
+    },
+    legend: {
+      visible: true,
+    },
+  };
+
+  if (monitorMetrics) {
+    const assets = [];
+
+    for (const metric of monitorMetrics) {
+      const asset = {
+        assetId: metric.assetId,
+        properties: [
+          {
+            aggregationType: defaultAggregationType, // Monitor has no aggregationType and appliation defaults to AVERAGE
+            propertyId: metric.propertyId,
+            resolution: convertResolution(metric.resolution),
+          },
+        ],
+      };
+
+      assets.push(asset);
+    }
+
+    const queryConfig = {
+      queryConfig: {
+        source: 'iotsitewise',
+        query: {
+          properties: [],
+          assets,
+        },
+      },
+    };
+
+    return {
+      ...defaultProperties,
+      ...queryConfig,
+    };
+  }
+  return defaultProperties;
+};
+
+export const convertMonitorToAppDefinition = (
+  monitorDashboardDefinition?: string,
+): DashboardDefinition => {
+  if (monitorDashboardDefinition) {
+    const definition = JSON.parse(
```

Comment:
error handling; cannot expect the definition can be safely parsed into a JSON. I suggest breaking this logic into 2 parts to keep migration logic "simple":
1. parsing of dashboard definition string + error handling
2. simple migration logic

Reply:
I was thinking if there was an error parsing the dashboard definition it would bubble up to [here](https://github.com/awslabs/iot-application/blob/rc/apps/core/src/migration/migration.service.ts#L222) and return the parsing error to the client which I believe is what we want. Do you think I should add a different error message here?

Reply:
ok to not break down the logic;
a message to help customer identify and resolve the error would be necessary to unblock the migration 👍

Reply:
Sounds good, I will add this in a followup PR. 

As we discussed offline, we want to be able to allow partial completion of the migration if some dashboards fail. This is because we don't want to prevent customers from migrating 90% of their dashboards if there is some edge cases that cause migration for the remaining 10% to fail. So in my update I'm going to track each dashboard definition parsing failure and return those failures at the end of the migration with a message like `there was an error parsing dashboard definitions for the following dashboards: <dashboardNames>`.